src/net/virtual_connection.rs

```diff
@@ -232,11 +263,13 @@ impl VirtualConnection {
         }
     }
 
-    /// This processes the incoming data and returns a packet if the data is complete.
+    // TODO would be super nice to return iterator of new packets, instead of passing `sender`, and `socket_manager` inside.
```

Comment:
We try to avoid TODO comments in laminar. So once this is merged, create an issue for this.

Reply:
I was planning to do exactly that.